openmc/mesh.py

```diff
@@ -433,6 +433,9 @@ def lower_left(self, lower_left):
         cv.check_length('mesh lower_left', lower_left, 1, 3)
         self._lower_left = lower_left
 
+        if self.upper_right is not None and any(np.isclose(self.upper_right, lower_left)):
```

Comment:
I like this PR and was taking a look at the np.isclose function. It has default tolerances, is it worth setting the tolerance to 0 in this call
```suggestion
        if self.upper_right is not None and any(np.isclose(self.upper_right, lower_left, rtol=0, atol=0)):
```